### PR TITLE
test: [M3-6960] - Add regression test for downloading invoice of maximum size

### PR DESCRIPTION
**Pull request by cliu-akamai**
## Description 📝
Add regression tests for downloading the invoice of maximum size.

## Major Changes 🔄
- Add regression tests to download the invoice with maximum size.
- Add `fs` library to `devDependency`.
- Add new setup plugins `downloadProcess` to clean up the download folder.

## How to test 🧪
```
yarn cy:run -s "cypress/e2e/core/billing/billing-invoices.spec.ts"
```

### CODE REVIEW

File: packages/manager/package.json [hunk 1]
```diff
@@ -191,6 +191,7 @@
     "eslint-plugin-testing-library": "^3.1.2",
     "eslint-plugin-xss": "^0.1.10",
     "factory.ts": "^0.5.1",
+    "fs": "^0.0.1-security",
```
Comment (jdamore-linode) [inline packages/manager/package.json:194]: I don't think we should have to add any packages to use the `fs` module (and it looks like the NPM [fs package](https://www.npmjs.com/package/fs) is basically a stub).

Did you get an error when you tried to use it? If so, I'm wondering if we need to include `node` in the [Cypress TypeScript config's](https://github.com/linode/manager/blob/develop/packages/manager/cypress/tsconfig.json#L23) `types` field.

File: packages/manager/cypress/support/plugins/download-processor.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        try {
+          readdirSync(folderPath).forEach((file) => {
+            rmSync(`${folderPath}/${file}`, { recursive: true });
+          });
+          resolve(null);
+        } catch (err) {
+          reject(err);
+        }
+      });
+    },
```
Comment (jdamore-linode) [inline packages/manager/cypress/support/plugins/download-processor.ts:21]: This makes a lot of sense since the downloaded invoice comes to a whopping 85MB on my machine -- that would add up really quickly.

From my experimenting, it looks like the downloaded invoice has the same filename every time thanks to your mocks, and I think Cypress's download behavior automatically overwrites any existing files so I'm not sure we have to worry about these invoices accumulating. Is there another  situation where this might be required?

File: packages/manager/package.json (hunk 1 above)
Comment (cliu-akamai) [reply packages/manager/package.json:194]: No, I don't have any issue when using it. Did you?

File: packages/manager/cypress/support/plugins/download-processor.ts (hunk 2 above)
Comment (cliu-akamai) [reply packages/manager/cypress/support/plugins/download-processor.ts:21]: Also, it might be helpful for other tests that will download items.

File: packages/manager/cypress/e2e/core/billing/billing-invoices.spec.ts [hunk 3]
```diff
@@ -401,4 +416,127 @@ describe('Account invoices', () => {
...
     });
   });
+
+  /*
+   * - Confirms that invoice with maximum size can be downloaded correctly.
+   */
+  it('can download an invoice with maximum size correctly from billing landing page', () => {
+    const mockInvoice = invoiceFactory.build();
+    const mockInvoiceItems = invoiceItemFactory.buildList(maximumInvoiceSize);
+    // mockInvoice.date = "2022-08-01T03:00:12";
```
Comment (carrillo-erik) [inline packages/manager/cypress/e2e/core/billing/billing-invoices.spec.ts:426]: Do we need to keep this comment here?

File: packages/manager/package.json (hunk 1 above)
Comment (carrillo-erik) [reply packages/manager/package.json:194]: As an alternative there is also the [fs-extra](https://www.npmjs.com/package/fs-extra) package which adds extra functionality.

**Comment by carrillo-erik** [comment]
I was able to run the tests locally without issues. All 6 tests are passing ✅

Screenshot 2023-11-08 at 3 25 15 PM

File: packages/manager/cypress/e2e/core/billing/billing-invoices.spec.ts (hunk 3 above)
Comment (cliu-akamai) [reply packages/manager/cypress/e2e/core/billing/billing-invoices.spec.ts:426]: Good catch, we don't.

**Review by mjac0bs — APPROVED** [review]
I had no issues running the billing-invoice.spec.ts locally; it passed consistently. 

I also had a look at the last e2e run. 
`smoke-domain-download-zone-file.spec.ts` passes locally, but failed in CI.
`billing-invoice.spec.ts` still fails in CI.
`clone-linode.spec.ts` was failing last week and I think this would be fixed if you merged the latest `develop` in.

It seems like the two download file specs could be failing due to what Joe mentioned about the lack of a download directory?
> It looks like this is also behind the 2 test failures in CI -- it's basically tripping up here because the downloads directory doesn't exist when this hook is called. That can be fixed pretty easily if we decide this clean up is required!

I'm approving pending that's addressed.

**Review by jaalah-akamai — APPROVED** [review]
Tests passed for me - @cliu-akamai if you accept suggested changes, let's just make sure things are good throughout the app 👍

File: packages/manager/cypress/e2e/core/billing/billing-invoices.spec.ts [hunk 4]
```diff
@@ -346,4 +361,126 @@ describe('Account invoices', () => {
...
+      const invoiceDateWithTime = formatDate(mockInvoice.date, {
+        timezone,
+        displayTime: true,
+      });
+
+      console.log('Invoice date: ' + invoiceDate);
+      console.log(
+        'Invoice date with time: ' +
+          invoiceDateWithTime.replace(' ', 'T').replace(':', '_')
+      );
```
Comment (coliu-akamai) [inline packages/manager/cypress/e2e/core/billing/billing-invoices.spec.ts:418]: same here

**Review by coliu-akamai — APPROVED** [review]
confirmed that tests locally passed on my end! ✅ 
saw the latest ci run is failing -- unfortunately I can't see what's causing them, so approving this pr pending that they're unrelated to the changes here!

File: packages/manager/cypress/e2e/core/billing/billing-invoices.spec.ts (hunk 4 above)
Comment (cliu-akamai) [reply packages/manager/cypress/e2e/core/billing/billing-invoices.spec.ts:418]: No we don't. I will remove them.

**Comment by jdamore-linode** [comment]
> @jdamore-linode @cliu-akamai just circling back on this PR - is this good to go?

@jaalah-akamai I'll try to circle back to this soon, but I think there may be changes we'll want to make to this before merging (cc @cliu-akamai -- I'll keep you posted too). For now I'm going to add the Wait to Merge label.
